src/Exercism.Representers.CSharp/Normalization/NormalizeDictionaryInitialization.cs

```diff
@@ -0,0 +1,180 @@
+                    };
+
+                if (!initializerExtractionResults.Success)
+                {
+                    return DefaultVisit();
+                }
+
+                var visitedInitializerSyntaxNodePairs = initializerExtractionResults.InitializerSyntaxNodePairs
+                    .Select(initializer => new KeyValuePair<SyntaxNode, SyntaxNode>(
+                        this.Visit(initializer.Key),
+                        this.Visit(initializer.Value))
+                    );
+
+                var replacementNode
+                    = BuildReplacementSyntaxWithCollectionInitialization(visitedInitializerSyntaxNodePairs);
+
+                if (replacementNode == default)
+                {
+                    var message
+                        = $"{nameof(NormalizeDictionaryInitialization)}: failed to find a {nameof(InitializerExpressionSyntax)} node in generated dictionary fragment";
+                    throw new NullReferenceException(message);
+                }
+
+                return base.VisitInitializerExpression(replacementNode as InitializerExpressionSyntax);
+            }
+            catch (InvalidKindException ike)
+            {
+                Log.Error(
+                    $"{nameof(NormalizeDictionaryInitialization)}: {nameof(InitializerExpressionSyntax)} found with unexpected Kind {ike.Kind}");
+                throw;
+            }
+            catch (Exception e)
+            {
+                Log.Error(e, $"{nameof(NormalizeDictionaryInitialization)}: unknown error");
+                throw;
+            }
+        }
+
+        private bool IsDictionary(InitializerExpressionSyntax initializerExpression)
+        {
+            var isDictionary = initializerExpression?.Parent?.ChildNodes()?.FirstOrDefault()?.GetText()?.ToString()
+                ?.StartsWith("Dictionary");
+            if (!isDictionary.HasValue)
+            {
+                Log.Error(
+                    $"{nameof(NormalizeDictionaryInitialization)}: unable to retrieve the type information for {nameof(initializerExpression)}");
+                return false;
+            }
+
+            if (!isDictionary.Value)
+            {
+                return false;
+                // some other sort of object
+            }
+
+            return true;
+        }
+
+        private (bool Success, List<KeyValuePair<SyntaxNode, SyntaxNode>> InitializerSyntaxNodePairs)
+            ExtractInitializersWithObjectSyntax(InitializerExpressionSyntax initializerExpression)
+        {
+            ArgumentSyntax GetArguemntSyntax(SyntaxNode node)
+            {
+                if (!node.ChildNodes().Any())
+                {
+                    return null;
+                }
+
+                if (node.ChildNodes().First() is ArgumentSyntax)
+                {
+                    return node.ChildNodes().First() as ArgumentSyntax;
+                }
+
+                return GetArguemntSyntax(node.ChildNodes().First());
+            }
```

Comment:
This has the tests still passing:

```suggestion
            ArgumentSyntax GetArguemntSyntax(SyntaxNode node) =>
                node.DescendantNodes()
                    .OfType<ArgumentSyntax>()
                    .FirstOrDefault();
            }
```